File: app/src/main/java/com/attendance/data/source/remote/UserRemoteDataSource.java
package com.attendance.data.source.remote;

import rx.Observable;
import rx.Subscriber;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

/**
 * Created by peiqin on 3/5/2017.
 */

public class UserRemoteDataSource {

    private UserRemoteDataSource() {
    }

    private static class SingletonHolder {
        private static final UserRemoteDataSource INSTANCE = new UserRemoteDataSource();
    }

    public static UserRemoteDataSource getInstance() {
        return SingletonHolder.INSTANCE;
    }

    public void login(Subscriber<Boolean> subscriber, String username, String password, boolean isTeacher) {

        boolean result = false;
        if (username.equals("1") && password.equals("1") && isTeacher) {
            result = true;
        }
        Observable observable = Observable.just(result);
        observable.subscribeOn(Schedulers.io())
                .unsubscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(subscriber);

    }

}

File: app/src/main/kotlin/com.attendance/data/source/remote/UserRemoteDataSource.kt
package com.attendance.data.source.remote

import rx.Observable
import rx.Subscriber
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers

/**
 * Created by peiqin on 3/5/2017.
 */

class UserRemoteDataSource private constructor() {

    private object SingletonHolder {
        val INSTANCE = UserRemoteDataSource()
    }

    fun login(subscriber: Subscriber<Boolean>, username: String, password: String, isTeacher: Boolean) {

        var result = false
        if (username == "1" && password == "1" && isTeacher) {
            result = true
        }
        val observable = Observable.just(result)
        observable.subscribeOn(Schedulers.io())
                .unsubscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(subscriber)

    }

    companion object {

        val instance: UserRemoteDataSource
            get() = SingletonHolder.INSTANCE
    }

}